lib/rule-context.js

```diff
@@ -88,15 +111,39 @@ function RuleContext(ruleId, eslint, severity, options, settings, ecmaFeatures)
 
     /**
      * Passthrough to eslint.report() that automatically assigns the rule ID and severity.
-     * @param {ASTNode} node The AST node related to the message.
+     * @param {ASTNode|MessageDescriptor} nodeOrDescriptor The AST node related to the message or a message
+     *      descriptor.
      * @param {Object=} location The location of the error.
      * @param {string} message The message to display to the user.
      * @param {Object} opts Optional template data which produces a formatted message
      *     with symbols being replaced by this object's values.
      * @returns {void}
      */
-    this.report = function(node, location, message, opts) {
-        eslint.report(ruleId, severity, node, location, message, opts);
+    this.report = function(nodeOrDescriptor, location, message, opts) {
+
+        var descriptor,
+            fix = null;
+
+        // check to see if it's a new style call
+        if (arguments.length === 1) {
+            descriptor = nodeOrDescriptor;
+
+            // if there's a fix specified, get it
+            if (typeof descriptor.fix === "function") {
+                fix = descriptor.fix(new RuleFixer());
+            }
+
+            eslint.report(
+                ruleId, severity, descriptor.node,
+                descriptor.loc || descriptor.node.loc.start,
+                descriptor.message, descriptor.data, fix
```

Comment:
I had it wirh one argument on each line at first, then reorganized to this because I thought it looked weird.